packages/engine/lib/stateful/src/agent/arrow/iterator.rs

```diff
@@ -56,11 +56,17 @@ pub fn json_value_iter_cols<'b: 'a, 'a>(
 pub fn index_iter<'b: 'a, 'a>(
     agent_pool: &'a [&'b AgentBatch],
 ) -> impl Iterator<Item = AgentIndex> + 'a {
-    agent_pool.iter().enumerate().flat_map(|(i, g)| {
-        let num_agents = g.num_agents() as u32;
-        let group_index = i as u32;
-        (0..num_agents).map(move |j| (group_index, j))
-    })
+    agent_pool
+        .iter()
+        .enumerate()
+        .flat_map(|(agent_batch_index, agent_batch)| {
```

Comment:
I don't think I mind us calling this the `group_index` here. The `batch_index` _is_ the `group_index`, e.g. if you `.zip()` agent_state and messages, the pairs are the groups and the first pair will have group_index 0.